### PR TITLE
inputs validation during signup

### PR DESCRIPTION
**Pull request by marioportela1993**
## Summary

sign up validations

## Trello Card

https://trello.com/c/yM6iZ3Tp/24-sign-up-validaciones

### CODE REVIEW

File: app/graphql/directives.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable class-methods-use-this */
+const { gql } = require('apollo-server');
+const { SchemaDirectiveVisitor } = require('graphql-tools');
+const { defaultFieldResolver } = require('graphql');
+const getValidationFn = require('../utils/validationFunctions');
+const errors = require('../errors');
+const logger = require('../logger');
```
Comment (lnmedrano) [inline app/graphql/directives.js]: 'const' can be explicit in the first variable. Then separate the others with comas.
const a = foo,
 b = foo;

File: app/graphql/directives.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+class ValidateInputDirective extends SchemaDirectiveVisitor {
+  visitFieldDefinition(field) {
+    const { resolve = defaultFieldResolver } = field;
```
Comment (lnmedrano) [inline app/graphql/directives.js]: Idem https://github.com/wolox-training/mpg-graphql/pull/7/files#r311113527

File: app/graphql/directives.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const { inputs } = this.args;
```
Comment (lnmedrano) [inline app/graphql/directives.js]: idem https://github.com/wolox-training/mpg-graphql/pull/7/files#r311113527

File: app/utils/users.js [hunk 4]
```diff
@@ -9,3 +9,15 @@ exports.encryptPassword = password =>
     logger.error(err.message);
     throw errors.hashError('Error hashing password');
   });
+
+exports.validateEmail = email => {
+  if (/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
+    const domain = email.split('@')[1];
+    if (domain === 'wolox.co' || domain === 'wolox.com.ar') {
```
Comment (FrancoCoronel) [inline app/utils/users.js]: directly you can do this:
```.js
return domain === 'wolox.co' || domain === 'wolox.com.ar'
```
```

File: app/graphql/directives.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable class-methods-use-this */
+const { gql } = require('apollo-server'),
+  { SchemaDirectiveVisitor } = require('graphql-tools'),
+  { defaultFieldResolver } = require('graphql'),
+  validateSchema = require('../utils/schemaValidator'),
+  logger = require('../logger');
+
+class ValidateSchemaDirective extends SchemaDirectiveVisitor {
```
Comment (gonziesc) [inline app/graphql/directives.js]: interesting... im not sure we should be using this kind of implementations, can be a little dark if they grow, but it's nice what you've done for this case

**Pull request merged**